test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -302,6 +305,16 @@ private MockTransportService(
             Tracer.NOOP
         );
         this.original = transport.getDelegate();
+        this.testExecutor = EsExecutors.newScaling(
+            "mock-transport",
+            0,
+            4,
+            30,
+            TimeUnit.SECONDS,
+            false,
```

Comment:
Could we set `rejectAfterShutdown` to `true` please?

Reply:
> it is not rejected and we should just queue it up

Sure but that's not what `false` does, it just silently drops any work submitted after shutdown.

My thinking here is twofold: firstly, we are terminating this `ExecutorService` at the end of the test, there should be no more tasks submitted by that point anyway. Secondly, IMO the silent-drop behaviour of `generic` after shutdown is a bug, even if it is unfortunately one we cannot easily address without changing lots of other things, but we can at least avoid relying on that same behaviour in specific situations like this one.

Reply:
> but that's not what false does, it just silently drops any work submitted after shutdown.

TIL. Thanks!